src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands/NewAzureContainerRegistryReplication.cs

```diff
@@ -0,0 +1,91 @@
+    [OutputType(typeof(PSContainerRegistryReplication))]
+    public class NewAzureContainerRegistryReplication : ContainerRegistryCmdletBase
+    {
+        [Parameter(Position = 0, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(Position = 1, Mandatory = true, ParameterSetName = NameResourceGroupParameterSet, HelpMessage = "Container Registry Name.")]
+        [Alias(ContainerRegistryNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string RegistryName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = RegistryObjectParameterSet, HelpMessage = "Container Registry Object.")]
+        [ValidateNotNull]
+        public PSContainerRegistry Registry { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Container Registry Location. Default to the location of the resource group.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ReplicationLocationAlias)]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Container Registry Replication Name. Default to the location name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ReplicationNameAlias, ResourceNameAlias)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Container Registry Tags.")]
+        [ValidateNotNull]
+        [Alias(TagsAlias)]
+        public Hashtable Tag { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = ResourceIdParameterSet, ValueFromPipelineByPropertyName = true, HelpMessage = "The container registry resource id")]
+        [ValidateNotNullOrEmpty]
+        [Alias(ResourceIdAlias)]
+        public string ResourceId { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            if (string.Equals(ParameterSetName, RegistryObjectParameterSet))
+            {
+                ResourceGroupName = Registry.ResourceGroupName;
+                RegistryName = Registry.Name;
+            }
+            else if (MyInvocation.BoundParameters.ContainsKey("ResourceId") || !string.IsNullOrWhiteSpace(ResourceId))
+            {
+                string resourceGroup, registryName, childResourceName;
+                if (!ConversionUtilities.TryParseRegistryRelatedResourceId(ResourceId, out resourceGroup, out registryName, out childResourceName))
+                {
+                    WriteInvalidResourceIdError(InvalidRegistryResourceIdErrorMessage);
+                    return;
+                }
+
+                ResourceGroupName = resourceGroup;
+                RegistryName = registryName;
+            }
+
+            var tags = TagsConversionHelper.CreateTagDictionary(Tag, validate: true);
+            if (string.IsNullOrEmpty(Name))
+            {
+                Name = Location.Replace(" ", string.Empty).ToLower();
```

Comment:
@Jinhuafei if the default behavior on the portal is to use the location, what happens if there is already a replication object named "WestUS"?

Reply:
If there is already a replication in `East US` but is called `westus` (very odd but it's allowed and possible). Then if the user goes to Portal/Az CLI/Powershell and tries to create a replication in `West US` without providing a name (so the name will default to `westus`), it will just fail because the name already exists.

I don't think there will be any concerns with Powershell users as they can always rerun the command with a different name such as `realwestus` in `West US` again. The same problem exists regardless if `name` is mandatory or optional here.

For Azure Portal, there was a different set of usability discussions that location has been used as replication name.